roles/rebuild-undercloud/tasks/main.yml

```
@@ -1,5 +1,26 @@
 ---
-# tasks file for rebuild-undercloud
+- name: Get the instackenv content for this cloud
+  uri:
+    url: "{{ instackenv_json }}"
+  register: uri_response
+
+- name: Count the number of nodes in instackenv
+  set_fact:
+    instackenv_nodes: "{{ uri_response['json']['nodes']|length }}"
+
+- name: Requesting {{ total_hosts }} from instackenv {{ instackenv_nodes }} nodes
+  fail:
+    msg: "The number of nodes requested {{ total_hosts }} is more hosts in instackenv file {{ instackenv_nodes }}."
```

I think this is a great idea.  We might want to make a new role for something like this such as `cloud_build_check` if we think we could make a few more "validation" checks so we can fail fast if a parameter can easily be determined to be incorrect.